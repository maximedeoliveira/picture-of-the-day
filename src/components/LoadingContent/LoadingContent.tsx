import React from 'react';
import { ActivityIndicator } from 'react-native-paper';

import Container from '@/components/Container/Container';

const LoadingContent = () => {
  return (
    <Container justifyContent="center" alignItems="center">
      <ActivityIndicator testID="loader" />
    </Container>
  );
};

export default LoadingContent;
