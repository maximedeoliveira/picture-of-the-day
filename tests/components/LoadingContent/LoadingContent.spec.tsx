import React from 'react';

import LoadingContent from '@/components/LoadingContent/LoadingContent';

import { render } from '../../utils';

describe('LoadingContent', () => {
  it('render component', () => {
    const component = render(<LoadingContent />);

    expect(component).toMatchSnapshot();
  });
});
