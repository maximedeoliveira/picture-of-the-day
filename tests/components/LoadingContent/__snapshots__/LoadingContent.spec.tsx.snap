// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContent render component 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "off",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "flex": 1,
      },
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "justifyContent": "center",
        "paddingHorizontal": 12,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      {
        "busy": true,
      }
    }
    accessible={true}
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    testID="loader"
  >
    <View
      collapsable={false}
      style={
        {
          "height": 24,
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 24,
              "transform": [
                {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 12,
                "overflow": "hidden",
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 24,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                    {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "height": 24,
              "transform": [
                {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 12,
                "overflow": "hidden",
                "top": 12,
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 24,
                  "transform": [
                    {
                      "translateY": -12,
                    },
                    {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "borderColor": "rgba(103, 80, 164, 1)",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
